presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkRddFactory.java

```diff
@@ -381,11 +389,50 @@ private PrestoSparkTaskSourceRdd createTaskSourcesRdd(
 
     private static SetMultimap<Integer, ScheduledSplit> assignSourceDistributionSplits(Session session, List<ScheduledSplit> splits)
     {
-        int taskCount = getSparkInitialPartitionCount(session);
-        checkArgument(taskCount > 0, "taskCount must be greater then zero: %s", taskCount);
         ImmutableSetMultimap.Builder<Integer, ScheduledSplit> result = ImmutableSetMultimap.builder();
-        for (int splitIndex = 0; splitIndex < splits.size(); splitIndex++) {
-            result.put(splitIndex % taskCount, splits.get(splitIndex));
+
+        boolean autoTunePartitionCount = isAutoTuneSparkPartitionCount(session);
+        autoTunePartitionCount = autoTunePartitionCount && !splits.stream()
+                .anyMatch(split -> !split.getSplit().getConnectorSplit().getSplitSizeInBytes().isPresent());
```

Comment:
nit: should this be `allMatch`? -- Since if some split has size, some other splits don't, we still cannot do auto tune right?

Reply:
I removed double negation to improve readability.